Chelone.py

```diff
@@ -0,0 +1,240 @@
+	for key in pressed_keys:
+		if key not in caught_keys and\
+		  key not in caught_keys_prev and\
+		  key not in caught_keys_prev_prev:
+
+			pressed_keys.remove(key)
+
+	for key in caught_keys:
+		if key not in pressed_keys:
+			pressed_keys.append(key)
+
+	caught_keys_prev_prev = caught_keys.copy()
+
+def on_key_press(event):
+	global caught_keys
+	caught_keys.append(event.keysym)
+
+
+def on_key_release(event):
+	global caught_keys
+	caught_keys.remove(event.keysym)
+
+
+
+class Sprite():
+	"""Object that contains data about the sprite. Also can be used to have code ran on each frame"""
+	x = 0
+	y = 0
+	var = {}
+	image = None
+	image_tk = None
+	parent_canvas = None
+	updateFunc = None
+	_passedUpdateFunc=None
+	
+	def __init__(self, image:Union[str, tk.PhotoImage], x:int=0, y:int=0, updateFunc:Callable[..., None]=lambda a:None, setupFunc:Callable[..., None]=lambda a:None):
+		self.x = x
+		self.y = y
+		self.var = {}
+
+		if isinstance(image, str):
+			self.image = tk.PhotoImage(file=image)
+		else:
+			self.image = image
+		self.image_tk = None
+		self.parent_canvas = None
+		self._passedUpdateFunc = updateFunc
+		self.updateFunc = updateFunc
+
+		setupFunc(self)
+
+	def move(self, x:int, y:int):
+		self.x += x
+		self.y += y
+		self.parent_canvas.move(self.image_tk, x, y)
+
+	def change_image(self, image:tk.PhotoImage):
+		self.image = image
+		self.parent_canvas.itemconfig(self.image_tk, image=self.image)
+
+	def shedule_anim(self, anim_frames:list):
+		global clear_img
+
+		i = 0
+		tmp = self._passedUpdateFunc
+		
+		def anim_sheduler(sprite):
+			nonlocal i, tmp
+
+			sprite.change_image(anim_frames[i])
+			i+=1
+
+			if i == len(anim_frames):
+				i = 0
+
+			tmp(sprite)
+
+		self.updateFunc = anim_sheduler
+
+class SpriteRenderer():
+	"""Main drawing class. Handles drawing every sprite"""
+	_root = None
+	_screen = None
+	_sprites = []
+	_prev_draw_time = None
+
+	def __init__(self, root: tk.Tk, screen:tk.Canvas):
+		self._root = root
+		self._screen = screen
+		
+		# 50 different z layers
+		self._sprites = []
+		for i in range(50):
+			self._sprites.append([])
+
+	def draw_gui(self):
+		pass
+
+
+	frame_period=1.0/TARGET_FPS
+	now=time()
+	next_frame=now+frame_period
+
+	def draw_scene(self):
+
+		while self.now<self.next_frame:
+			sleep(self.next_frame-self.now)
+			self.now=time()
+
+		for layer in self._sprites:
+			for sprite in layer:
+
+				sprite.updateFunc(sprite)
+
+
+		self._root.update()
+		self.next_frame+=self.frame_period
+
+	def move_camera(self, x, y):
+		self._screen.move("all",x,y)
+
+	def add_sprite(self, sprite:Sprite, layer:int = 25):
+		
+		#layer 0 reserved for gui, layer 50 onward does not exist
+		if layer < 1 or layer > 49:
+			return
+
+		self._sprites[layer].append(sprite)
+
+		sprite.image_tk = self._screen.create_image(sprite.x, sprite.y, anchor=tk.NW, image=sprite.image)
+		sprite.parent_canvas = self._screen
+
+		for i, z_layer in enumerate(self._sprites):
+			if len(z_layer)>0 and i<=layer:
```

Comment:
comparison operator  must be surronded by a single space

Reply:
Done